tests/ui/codegen/duplicated-path-in-error.rs

```diff
@@ -1,5 +1,6 @@
 //@ only-linux
 //@ compile-flags: -Zcodegen-backend=/non-existing-one.so
+//@ normalize-stderr: "Error loading shared library .+:" -> "cannot open shared object file:"
 
 // This test ensures that the error of the "not found dylib" doesn't duplicate
 // the path of the dylib.
```

Comment:
then on the ERROR down here we will prefix it. the current one will be `//[gnu]` and the new one will be `//[musl]`, containing the messages we are expecting, so like this:
```suggestion
// the path of the dylib.

// other stuff I can't comment on because GitHub reviews aren't great

//[gnu]~? ERROR couldn't load codegen backend /non-existing-one.so
//[musl]~? ERROR idk, you're the one making the PR
```

Reply:
Nice, thanks for the hints! I think I got it working as intended now

Reply:
I think this is actually fine because the prefix message is the same, and now we're revisioning gnu vs musl error messages, but this needs a comment describing why there are `only-gnu` vs `only-musl` revisions in the first place.

Reply:
The original test was an anti-regression for glibc's duplicate path error message, so I'm fine with limiting it to gnu-only.

Reply:
Should I do that instead then or leave it as it is now?

Reply:
Yeah, the current version is fine.